clar2wasm/src/words/blockinfo.rs

```diff
@@ -53,3 +53,314 @@ impl Word for GetBlockInfo {
         Ok(())
     }
 }
+
+#[derive(Debug)]
+pub struct GetBurnBlockInfo;
+
+impl Word for GetBurnBlockInfo {
+    fn name(&self) -> ClarityName {
+        "get-burn-block-info?".into()
+    }
+
+    fn traverse(
+        &self,
+        generator: &mut WasmGenerator,
+        builder: &mut walrus::InstrSeqBuilder,
+        expr: &SymbolicExpression,
+        args: &[SymbolicExpression],
+    ) -> Result<(), GeneratorError> {
+        let prop_name = args.get_name(0)?;
+        let block = args.get_expr(1)?;
+
+        // Push the property name onto the stack
+        let (id_offset, id_length) = generator.add_string_literal(prop_name);
+        builder
+            .i32_const(id_offset as i32)
+            .i32_const(id_length as i32);
+
+        // Push the block number onto the stack
+        generator.traverse_expr(builder, block)?;
+
+        // Reserve space on the stack for the return value
+        let return_ty = generator
+            .get_expr_type(expr)
+            .expect("get-burn-block-info? expression must be typed")
+            .clone();
+
+        let (return_offset, return_size) =
+            generator.create_call_stack_local(builder, &return_ty, true, true);
+
+        // Push the offset and size to the data stack
+        builder.local_get(return_offset).i32_const(return_size);
+
+        // Call the host interface function, `get_burn_block_info`
+        builder.call(generator.func_by_name("get_burn_block_info"));
+
+        // Host interface fills the result into the specified memory. Read it
+        // back out, and place the value on the data stack.
+        generator.read_from_memory(builder, return_offset, 0, &return_ty);
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub struct AtBlock;
+
+impl Word for AtBlock {
+    fn name(&self) -> ClarityName {
+        "at-block".into()
+    }
+
+    fn traverse(
+        &self,
+        generator: &mut WasmGenerator,
+        builder: &mut walrus::InstrSeqBuilder,
+        _expr: &SymbolicExpression,
+        args: &[SymbolicExpression],
+    ) -> Result<(), GeneratorError> {
+        let block_hash = args.get_expr(0)?;
+        let e = args.get_expr(1)?;
+
+        // Traverse the block_hash, leaving it on the top of the stack
+        generator.traverse_expr(builder, block_hash)?;
+
+        // Call the host interface function, `enter_at_block`
+        builder.call(generator.func_by_name("enter_at_block"));
+
+        // Traverse the inner expression
+        generator.traverse_expr(builder, e)?;
+
+        // Call the host interface function, `exit_at_block`
+        builder.call(generator.func_by_name("exit_at_block"));
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use clarity::vm::{
+        types::{OptionalData, PrincipalData, TupleData},
+        Value,
+    };
+
+    use crate::tools::{evaluate, TestEnvironment};
+
+    // These tests are disabled because they require a block to be present in the
```

Comment:
Are they though? Is this comment outdated or should it say "some of these tests?"

Reply:
Deleted this comment in 47c62bb57ed615d6f5dd5ecffdca2731cee83f56